--- serverloop.c-orig	2022-08-26 12:41:27.675690200 -0400
+++ serverloop.c	2022-08-27 13:59:30.317424200 -0400
@@ -831,8 +831,7 @@
 		/* check permissions */
 		if ((options.allow_streamlocal_forwarding & FORWARD_REMOTE) == 0
 		    || !auth_opts->permit_port_forwarding_flag ||
-		    options.disable_forwarding ||
-		    (pw->pw_uid != 0 && !use_privsep)) {
+		    options.disable_forwarding) {
 			success = 0;
 			ssh_packet_send_debug(ssh, "Server has disabled "
 			    "streamlocal forwarding.");
